UnityProject/Assets/Scripts/Objects/Engineering/RadiationCollector.cs

```diff
@@ -0,0 +1,177 @@
+﻿using System;
+using UnityEngine;
+
+namespace Objects.Engineering
+{
+	public class RadiationCollector : MonoBehaviour, ICheckedInteractable<HandApply>
+	{
+		private ElectricalNodeControl electricalNodeControl;
+		private ModuleSupplyingDevice moduleSupplyingDevice;
+		private PushPull pushPull;
+		private RegisterTile registerTile;
+
+		private bool isOn;
+		private bool isWrenched;
+		private GameObject slotObject;
+
+		[SerializeField]
+		private SpriteHandler mainSpriteHandler = null;
+		[SerializeField]
+		private SpriteHandler slotHandler = null;
+
+		[SerializeField]
+		[Tooltip("Whether this radiation collector should start wrenched")]
+		private bool startSetUp;
+
+		[SerializeField]
+		[Tooltip("radiationWatts * radiation tile level = watts this device will supply")]
+		private float radiationWatts = 100f;
+
+		#region LifeCycle
+
+		private void Awake()
+		{
+			electricalNodeControl = GetComponent<ElectricalNodeControl>();
+			moduleSupplyingDevice = GetComponent<ModuleSupplyingDevice>();
+			pushPull = GetComponent<PushPull>();
+			registerTile = GetComponent<RegisterTile>();
+		}
+
+		private void OnEnable()
+		{
+			UpdateManager.Add(CollectorUpdate, 1f);
+		}
+
+		private void OnDisable()
+		{
+			UpdateManager.Remove(CallbackType.PERIODIC_UPDATE, CollectorUpdate);
+		}
+
+		private void Start()
+		{
+			if(CustomNetworkManager.IsServer == false) return;
+
+			if (startSetUp)
+			{
+				isWrenched = true;
+				pushPull.ServerSetPushable(false);
+			}
+		}
+
+		#endregion
+
+		private void CollectorUpdate()
+		{
+			if(CustomNetworkManager.IsServer == false) return;
+
+			MetaDataNode node = registerTile.Matrix.GetMetaDataNode(registerTile.LocalPositionServer);
+			if (node  == null) return;
+
+			moduleSupplyingDevice.ProducingWatts = node.RadiationNode.RadiationLevel * radiationWatts;
+		}
+
+		public bool WillInteract(HandApply interaction, NetworkSide side)
+		{
+			if (!DefaultWillInteract.HandApply(interaction, side)) return false;
+
+			if (Validations.HasItemTrait(interaction.HandObject, CommonTraits.Instance.Wrench)) return true;
+
+			if (interaction.HandObject == null) return true;
+
+			//TODO adding/removing tank logic
+
+			return false;
+		}
+
+		public void ServerPerformInteraction(HandApply interaction)
+		{
+			if (Validations.HasItemTrait(interaction.HandObject, CommonTraits.Instance.Wrench))
+			{
+				TryWrench(interaction);
+			}
+			else if (interaction.HandObject == null)
+			{
+				SupplyToggle(interaction);
+			}
+		}
+
+		private void TryWrench(HandApply interaction)
```

Comment:
I see there's just one step to unsecure the collector - wrenching. Perhaps the `WrenchSecurable` component would be suitable here?

Reply:
Theres a check before whether the radiation collector is off, wouldnt be able to do that with the component. Should it be allowed to be wrenched when turned on?

Reply:
You can disable the component when on. Up to you.

Reply:
Might leave it as it is for now, but i'll take another look when i add the fuel interaction in a different PR